### DISCUSSION TITLE
Python Beginner needs help :)

### DISCUSSION THREAD

**Discussion by DonFloriano27** [dcaribou/transfermarkt-datasets#63] [Q&A] [answered]
Hi everyone!
I'm working on my master thesis and need the data of the transfermarkt website. Your scraper seems to be the best way to collect it. 
In detail its the player data (name, age, nationality, current club, market-value of player, consulting agency) of all players of germans top 3 league clubs.
I have very basic python skills and a hard time to understand how this scraper will work.

I would highly appreciate any help! 

Thanks a lot :)
Flo

**Comment by dcaribou** [comment, marked as answer]
Hi @DonFloriano27. How can I help you?
I mean, most of the data you are referring to is already published in the project datasets in data.world and Kaggle in the `players` resource. Are you looking into enhancing the dataset with additional columns?

**Reply by DonFloriano27** [reply]
Hi @dcaribou,
Thanks for your quick reply!
Yes in the kaggle and data.world datasets I can find most of the desired data, only "player_agent" and "market_value" are missing. I can see the "player_agent" in the sample players.json of this repository so the scraper seems to extract it. "market_value" would be very important for me too and seems to be not scraped yet.
So yes, I am looking into enhancing the dataset with this two columns. Is this possible at all?

**Reply by dcaribou** [reply]
Got it.
Yes, what you want is very much possible and I welcome contributions 😄 

As you mentioned, the `player_agent` is already being scraped, so it should be quite easy to add to the dataset. You just need to set column in the `process_segment` function here:
https://github.com/dcaribou/transfermarkt-datasets/blob/6b6dd6572f582b2c40039913a65ba99d10fd1f44/prep/assets/clubs.py#L15

For the `market_value` it's a bit more work, since the scraping needs to be figured out. We'll need to update [this](https://github.com/dcaribou/transfermarkt-scraper/blob/cbb5e61a67ded7fcfc1e904f9401cbd90ba113f2/tfmkt/spiders/players.py#L35) function. Can you create an issue in the [transfermarkt-scraper](https://github.com/dcaribou/transfermarkt-scraper) repo? I'll help you getting started with the environment so you can the field.

Once the `market_value` is added to the scraper, you'll be able to set it in the dataset the same as for `player_agent`

**Reply by DonFloriano27** [reply]
Awesome! So can you help me to set up the environment?

**Reply by dcaribou** [reply]
Sure, let's follow-up in the issue.
